app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -300,4 +336,44 @@ public void showNotificationGroup(long priority) {
                 (NotificationManager) this.getSystemService(Context.NOTIFICATION_SERVICE);
         notificationManager.notify(-5, b.build());
     }
+
+    private Bitmap getIcon(Integer appid) {
+
+        if(appid==-1){
+            return BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
+        }
+
+        Bitmap icon;
+        RequestCreator request;
+        try {
+            request= cache.load(Utils.resolveAbsoluteUrl(settings.url() + "/", appIdMap.get(appid)));
+            icon = request.get();
+        } catch (IOException e) {
+            icon = BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
+        }
+
+        return icon;
+    }
+
+    private Picasso makePicasso() {
```

Comment:
Picasso is also created inside the message activity. Could you extract this method to com.github.gotify.Utils, to prevent code duplication?

Reply:
@jmattheis The makePicasso-functions are not identical. Generally i would advise againts having a Utils class since they often fill up with stuff that should be somewhere else.

In this case, would it be fine with you if i create a chache-handler class that provides the appropriate methods?

Reply:
Yeah, feel free to make a class for this.